src/state/slices/tradeQuoteSlice/selectors.ts

```diff
@@ -0,0 +1,89 @@
+import { createSelector } from '@reduxjs/toolkit'
+import { SwapperName } from 'lib/swapper/api'
+import { assertUnreachable } from 'lib/utils'
+import type { ReduxState } from 'state/reducer'
+import {
+  getHopTotalNetworkFeeFiatPrecision,
+  getHopTotalProtocolFeesFiatPrecision,
+  getInputOutputRatioFromQuote,
+  getNetReceiveAmountCryptoPrecision,
+  getTotalNetworkFeeFiatPrecision,
+  getTotalProtocolFeeByAsset,
+} from 'state/slices/tradeQuoteSlice/helpers'
+
+const selectTradeQuoteSlice = (state: ReduxState) => state.tradeQuoteSlice
+
+export const selectSelectedQuote = createSelector(selectTradeQuoteSlice, swappers => swappers.quote)
```

Comment:
```suggestion
export const selectSelectedQuote = createDeepEqualOutputSelector(selectTradeQuoteSlice, swappers => swappers.quote)
```

or better yet (for performance), cache based on an ID. Might need to add a TODO for this though because i dont think quotes have IDs attached

Reply:
https://github.com/shapeshift/web/pull/4782/commits/ff03b92e29699e757281d5948db6e783cb48ca52